modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/WeightedSampling.scala

```diff
@@ -0,0 +1,44 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.MonadThrow
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+
+object WeightedSampling {
+
+  type ProbabilityRefinement = Interval.ClosedOpen[0.0, 1.0]
+  type ProbabilityRefined = Double Refined ProbabilityRefinement
+
+  type WeightRefinement = Interval.OpenClosed[0.0, 1.0]
+  type WeightRefined = Double Refined WeightRefinement
+
+  def sample[F[_]: MonadThrow, A](getProbability: F[ProbabilityRefined])(distribution: List[(A, WeightRefined)]): F[A] = {
```

Comment:
why one object is named `WeightedSampling` and another `WeightedProspectSample`?
Shouldn't it be either `WeightedSample`/`WeightedProspectSample` or `WeightedProspect`/`WeightedSampling`?

Reply:
`WeightedSample` returns a single sample. `WeightedProspectSample` calls `WeightedSample` multiple times and returns a list of samples. `WeightedProspectSample` has been renamed to `WeightedProspect`.